### PR TITLE
[Feature] - CSS Inliner for Email

### PR DESCRIPTION
**Pull request by EduardoDePatta**
Hello!

This PR introduces a new `CSS to Inline Converter` page to the application, designed to help users easily convert CSS styles into inline styles within their HTML. This tool is especially useful for developers who need to improve email compatibility or reduce external stylesheet dependencies.

### Key Changes
1. **New Utility Function:**
   - A utility function was added to parse HTML and CSS, applying inline styles from the CSS to the corresponding elements in the HTML while preserving existing inline styles.
2. **Test Suite:**
   - Created comprehensive tests to validate the functionality of the `convertCSSToInline` function, ensuring it handles various scenarios such as conflicting styles, non-matching selectors, and empty inputs.
3. **New Page Component:**
   - Added a new page component that allows users to input HTML and CSS, convert the CSS to inline styles, and copy the result.
   - Includes input validation to ensure correct HTML is provided and handles errors.
   - Features copy-to-clipboard functionality for easily copying the inlined HTML output.
4. **Updated Tools List:**
   - The `tools` array was updated to include the new converter, making it accessible from the main tools page.
5. **Documentation Update:**
   - The README file was updated to include the new converter in the list of available utilities, with a direct link to the tool.

### Benefits
1. **Developer Efficiency:** This tool simplifies the process of converting CSS to inline styles, a common need for email template development and other scenarios where external stylesheets are not supported.

### CODE REVIEW

**Comment by peckz** [comment]
@EduardoDePatta Thank you for this amazing tool! Great idea and great execution! left few minor comments

File: components/utils/tools-list.ts [hunk 1]
```diff
@@ -83,4 +83,10 @@ export const tools = [
       "Easily convert numbers between different bases (binary, octal, decimal, hexadecimal) with our free online Number Base Changer.",
     link: "/utilities/number-base-changer",
   },
+  {
+    title: "CSS to Inline Converter",
+    description:
+      "Easily convert your CSS styles to inline styles directly in your HTML. Perfect for improving email compatibility and reducing external stylesheet dependencies.",
+    link: "/utilities/css-to-inline-converter",
```
Comment (peckz) [inline components/utils/tools-list.ts:90]: Once you update title, would be nice also to update other link instances with `css-inliner-for-email`

File: components/utils/tools-list.ts [hunk 2]
```diff
@@ -83,4 +83,10 @@ export const tools = [
       "Easily convert numbers between different bases (binary, octal, decimal, hexadecimal) with our free online Number Base Changer.",
     link: "/utilities/number-base-changer",
   },
+  {
+    title: "CSS to Inline Converter",
```
[the remark this answers is not in the document]
Comment (EduardoDePatta) [reply components/utils/tools-list.ts]: I've updated the title to "CSS Inliner for Email" as recommended. I appreciate the guidance!

File: pages/utilities/css-to-inline-converter.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    const errorMessage =
+      conditions.find((value) => value.condition)?.message || null;
+
+    try {
+      const inlinedHtml = convertCSSToInline(htmlInput, cssInput);
+      setOutput(errorMessage ?? inlinedHtml);
+    } catch (errorMessage: unknown) {
+      setOutput(errorMessage as string);
+    }
```
[the remark this answers is not in the document]
Comment (peckz) [reply pages/utilities/css-to-inline-converter.tsx]: Makes sense, thank you!

**Review by peckz — APPROVED** [review]

**Comment by peckz** [comment]
@EduardoDePatta Thank you once again and welcome to the contributors list!

**Comment by danigrant** [comment]
This is SO awesome!!! Such a great idea, I needed something like this so many times!

We'd love to send you a Jam hat to say thanks – my email is dani@jam.dev, if you reach out with your mailing address, a hat will be on its way! 🧢🤩

**Pull request merged**